core/modules/search/search.extender.inc

```diff
@@ -273,7 +275,12 @@ class SearchQuery extends SelectQueryExtender {
         foreach ($key as $or) {
           list($num_new_scores) = $this->parseWord($or);
           $any |= $num_new_scores;
-          $query_or->condition('d.data', "% $or %", 'LIKE');
+          if ($search_partial_words) {
+            $query_or->condition('d.data', "%$or%", 'LIKE');
+          }
+          else {
+            $query_or->condition('d.data', "% $or %", 'LIKE');
```

Comment:
This is a very simple solution, but is it going to provide the expected results? This isn't really stemming, so it's going to return a bunch of stuff you don't want in addition to the things you might want.